### PR TITLE
New quirks: Fighting Styles, Heat/Cold-Defying, O- Blood Type

### PR DESCRIPTION
**Pull request by Mickyan**
:cl: Mickyan
add: New major positive quirks: Unarmed Styles! Get ready to defend your dojo with 3 different techniques to choose from, each with its own pros and cons.
add: New positive quirks: Heat/Cold-defying. Which one do you like to hang around the most, space vacuum or plasma fires?
add: New neutral quirk: O- Blood Type. Medbay will surely appreciate you donations!
/:cl:

**Unarmed Styles:**
Simply make your unarmed strikes a bit stronger in different ways, and also add a cool custom attack message to show everyone how much of a badass you are. 

These are not strong enough to be a substitute to using a proper weapon, but they give you that extra edge in a pinch... or a bar brawl.

They cancel each other out, so only pick one.

**Available styles:**
_Boxer:_ Balanced style, pretty much like normal unarmed attacks, just a bit stronger.

_Martial Artist:_ Less damage than boxer but your damage numbers are more reliable and you're more likely to stun opponents (note that if you're just looking for a stun, disarm still has a much bigger chance)

_Slugger:_ You can hit really hard, sometimes. Your damage numbers are all over the place and you're the least likely to stun

Check out this handy chart I made while working out the balance, it shows the damage range, stun threshold and the final chance to stun:

(Slightly outdated, M.Art has a min damage of 3 with 14% stun chance)
fightingstylesbalance

**Heat/Cold-defying:**
Respectively makes you 20% more resistant to one and 10% more vulnerable to the opposite. You can stack them for diminishing returns at a higher cost (10% resistance to each for 2 points)

**O- Blood type:**
Your blood can be used for transfusions with anyone, but you can only get transfusions of your own blood type. Good luck with that.

### CODE REVIEW

**Comment by adamkad1** [comment]
o- blood type trait? no just no

**Comment by Tlaltecuhtli** [comment]
are those lines equally distributed? slugger seems overall shit as you have a minimum of 0

File: code/datums/traits/good.dm [hunk 1]
```diff
@@ -29,6 +29,21 @@
...
+	desc = "You're used to cold environments. You take less damage from intense cold, but slightly more damage from heat"
+	value = 1
+	gain_text = "<span class='notice'>You feel at ease in the cold.</span>"
+	lose_text = "<span class='notice'>You no longer feel at ease in the cold.</span>"
+
+/datum/quirk/cold_res/add()
+	var/mob/living/carbon/human/H = quirk_holder
+	var/datum/species/species = H.dna.species
+	species.coldmod = species.coldmod - 0.2
+	species.heatmod = species.heatmod + 0.1
```
Comment (XDTM) [inline code/datums/traits/good.dm]: use the physiology datum!

**Comment by davethwave** [comment]
Seems fine to me, but I don't really see a need to make the ability to have O- blood a trait.

**Comment by iskyp** [comment]
please do not make O- a quirk

**Comment by Mickyan** [comment]
@iskyp if you'd care to write up a reason for it maybe you can actually convince me

**Comment by iskyp** [comment]
should be a character creation option if we're going down the route of giving you control over BT, not a quirk, and even if we did add it to quirks, it's should either be all bloodtypes or none (which if it's 1 or more it becomes a hassle until we add in conflicting traits)
O- isn't even that special in ss13 because random blood types at roundstart



also that lose text literally is pointless because your blood type will already be set to O- and even if you lose the trait you'll still have special snowflake blood
another annoyance: it will appear on medical scanners on the quirks list even though medical scanners already show your blood type

File: code/datums/traits/good.dm [hunk 2]
```diff
@@ -39,6 +54,21 @@
...
+	desc = "You're used to hot environments. You take less damage from intense heat, but slightly more damage from cold"
+	value = 1
+	gain_text = "<span class='notice'>You feel at ease in the heat.</span>"
+	lose_text = "<span class='notice'>You no longer feel at ease in the heat.</span>"
+
+/datum/quirk/heat_res/add()
+	var/mob/living/carbon/human/H = quirk_holder
+	var/datum/species/species = H.dna.species
+	species.heatmod = species.heatmod - 0.2
+	species.coldmod = species.coldmod - 0.1
```
Comment (PKPenguin321) [inline code/datums/traits/good.dm]: am i reading this wrong or are one of these being applied incorrectly

**Comment by Iamgoofball** [comment]
>17% chance to stun on martial art

yeah so people are just only going to take martial art

**Comment by Spaceinaba** [comment]
slugger should never stun tbh
you're trading potential damage for potential stun chance

**Comment by Spaceinaba** [comment]
also for once yeah what goof said

File: code/datums/traits/good.dm (hunk 2 above)
Comment (Mickyan) [reply code/datums/traits/good.dm]: you're not wrong, last minute oversight

**Comment by epochayur** [comment]
yeah this sucks

Quirks work for me in theory for stuff like "yeah I want a robot arm", or "I wanna talk like an italian", or "I want to be anemic", but just "You are now straight-up better in (some) combat situations" seems stupid. I don't wanna have to metagame quirks (which should be an optional system in the first place) to be at the same level as everyone else in hand-to-hand combat.

And don't kid yourself, Martial Artist will be the only style anyone pics

**Comment by py01** [comment]
Boxer and slugger both have an average dph of 7.5, but boxing has +4% stun rate and more consistent damage. It seems pointless to include a perk that is strictly statistically worse than another perk, when they are mutually exclusive.

**Comment by PKPenguin321** [comment]
martial artist is absolutely the best one, yes

**Comment by iskyp** [comment]
of course i know how blood types work, O- isn't special in ss13 because a) blood is basically only useful to people with ABD and b) saline glucose is much easier to obtain and has the same effect

the difference between martial art and disarm is that martial art does damage which can slow people after they accumulate enough and disarm does no damage
you're trading 6% chance to stun for 4-9 damage per hit. Now, if you get maximum damage every hit, you'll get roughly 2 stuns in 12 hits.

**Comment by PKPenguin321** [comment]
also what about something like "limp-wristed" that lowers your punch damage

**Comment by Mickyan** [comment]
@epochayur Kor mentioned that none of the choices of quirks felt meaningful, this is my answer to that

If you don't want to metagame quirks you can pick up a toolbox and be one step above the idiot that's trying to kill you with his fists and had to choose a bunch of negative traits for it

@py01 slugger is luck based, basically. You can beat someone senseless in a couple of punches if you're very lucky but if you want consistent damage you'd go for boxer

@PKPenguin321 already nerfed martial arts a bit, stun is 14% with min damage at 3. Consider that someone with boxer can just switch to disarm for a much better chance to stun and then far more efficiently pummel you while you're down instead of spamming weak attacks with martial arts.

Maybe when/if we get an option to make mutually exclusive perks, I don't want to add even more ways to cheese with pacifist. Also given how disarming and using weapons is the de facto way to kill people I figured it wasn't much of a negative

**Comment by optimumtact** [comment]
no quirks that affect combat balance thanks.

It's just not worth further complicating an already fucked up system.

**Comment by optimumtact** [comment]
like a 5-10% resist buff would have been fine but 20% is imo absurd

**Comment by optimumtact** [comment]
in fact shit, none of these "quirks" are good, the o- blood trait is absurdly overpowered, the whole reason blood picks are randomized is to stop everyone being an o- blood type

**Comment by optimumtact** [comment]
I think you totally missed the point of the quirks system

**Comment by iskyp** [comment]
wouldn't O- be the least powerful while AB+ would be the most powerful

**Comment by Mickyan** [comment]
what is the point of the quirk system and who gets to define it

**Comment by epochayur** [comment]
literally the design lead and @Xhuis

**Comment by optimumtact** [comment]
no, just the design lead and the maintainers

**Comment by Mickyan** [comment]
I know, there was a discussion on the forums to make quirks more impactful to the round.

And on second thought I don't have the willpower to argue for it any longer right here and now so maybe we can talk about this again if they ever figure that out on their own.

**Comment by Therapist1** [comment]
O blood shouldn’t be a quirk, but I like the concept. The balance may need some tweaks, but I think this is a good idea.

**Comment by Xhuis** [comment]
@epochayur I do appreciate the consideration, but please don't ping me in this repo. I've decided to step away from /tg/ and its development.

**Comment by epochayur** [comment]
oh, bummer dude, you made some great stuff, godspeed

**Pull request closed without merging**